### ISSUE TITLE
Lambda Docker image format 409 with publish=true

### ISSUE THREAD

**Issue by wperron** [hashicorp/terraform-provider-aws#16907]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.14.3
+ provider registry.terraform.io/hashicorp/aws v3.22.0
```

### Affected Resource(s)

* aws_lambda_function

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# 8 Lambdas with configs similar to this one
# the only parts that differ are memory_size, timeout and environment variables
resource "aws_lambda_function" "stargazers" {
  package_type  = "Image"
  image_uri     = local.ecr_image_url
  function_name = "my_lambda_1"
  role          = aws_iam_role.lambda_exec_role.arn
  publish       = true
  timeout       = 300
  memory_size   = 1024

  image_config {
    command = ["main.handler"]
  }

  tags = local.tags
}
```

### Debug Output

https://gist.github.com/wperron/855c1b8a7f5e14815fe9c244127c624e

### Expected Behavior

I updated my Lambda functions to use the new Docker image format for deployments, and I started getting 409 errors due to conflicting operations when I updated all my functions. When I used a simple zip I didn't have this problem; it's only 8 functions so hitting this error now is a bit strange.

I tried using the `-parallelism` option to update the functions one at a time, but it seems to be ignored - all the functions start updating at the same time. I'm not sure if this part is a provider bug or a bug in the Terraform internals though.

### Actual Behavior

First I would expect that updating 8 lambdas concurrently not throw this error - it worked fine before switching to the Docker image format.

Second, if that's a new behavior I have to work around, I would expect the `-parallelism` option to actually limit the number of concurrent operations done by Terraform.

### Steps to Reproduce

1. `docker tag your-image:latest <account-id>.dkr.ecr.<region>.amazonaws.com/<your-repo>:latest-2` (just create a new tag from the same image
2. `docker push <account-id>.dkr.ecr.<region>.amazonaws.com/<your-repo>:latest-2`
3. `terraform apply`

### Important Factoids

N/A

### References

For a complete reference on the Terraform configuration, see https://github.com/denoland/deno_registry2

**Comment by wperron** [comment]
I went back to this issue, I tried to bypass it by setting `-parallelism=1` but it didn't work. I then tried to set explicit `depends_on` block to each of the lambda functions -- it _did_ fix the parallelism issue, but every function failed to publish with the error message "Error publishing Lambda Function [...] An update is in progress for resource".

So I tried updating the `publish` attribute from `true` to `false` and now I can deploy without issue.

Bottom line is; I think the issue has something to do with having `publish = true` while using the Docker format in a Lambda function
